examples/all-clusters-app/esp32/main/DeviceCallbacks.cpp

```diff
@@ -164,6 +168,36 @@ void DeviceCallbacks::OnLevelControlAttributeChangeCallback(EndpointId endpointI
     return;
 }
 
+// Current we only support ColorControl cluster for ESP32C3_DEVKITM which has an on-board RGB-LED
+#if CONFIG_DEVICE_TYPE_ESP32_C3_DEVKITM
+void DeviceCallbacks::OnColorControlAttributeChangeCallback(EndpointId endpointId, AttributeId attributeId, uint8_t * value)
+{
+    VerifyOrExit(attributeId == ZCL_COLOR_CONTROL_CURRENT_HUE_ATTRIBUTE_ID ||
+                     attributeId == ZCL_COLOR_CONTROL_CURRENT_SATURATION_ATTRIBUTE_ID,
+                 ESP_LOGI(TAG, "Unhandled AttributeId ID: '0x%04x", attributeId));
+    VerifyOrExit(endpointId == 1 || endpointId == 2, ESP_LOGE(TAG, "Unexpected EndPoint ID: `0x%02x'", endpointId));
+    if (endpointId == 1)
+    {
+        uint8_t hue, saturation;
+        if (attributeId == ZCL_COLOR_CONTROL_CURRENT_HUE_ATTRIBUTE_ID)
+        {
+            hue = *value;
+            emberAfReadServerAttribute(endpointId, ZCL_COLOR_CONTROL_CLUSTER_ID, ZCL_COLOR_CONTROL_CURRENT_SATURATION_ATTRIBUTE_ID,
+                                       reinterpret_cast<uint8_t *>(&saturation), sizeof(uint8_t));
```

Comment:
Include `app/common/gen/attributes/Accessors.h` and use:
```
  app::Clusters::ColorControl::Attributes::GetCurrentSaturation(endpointId, &saturation);
```
(or `using namespace chip::app::Clusters::ColorControl::Attributes` and then skip the prefix, of course).

Reply:
There are two issues after including the `app/common/gen/attributes/Accessors.h`
1. The `app/common/gen/attributes/Accessors.h`(or `app/common/gen/attributes/Accessors.cpp` ) doesn't include files `app/util/af.h` and `app/common/gen/attribute-type.h`.
2. The Attribute `LastConfiguredBy` of the cluster `Scenes` has an undefined type `ZCL_EUI64_ATTRIBUTE_TYPE`.

Reply:
@vivien-apple See above.

@wqx6 alright, let's not worry about that for now.  But in that case, can we at least remove the unnecessary `reinterpret_cast` bits?

Reply:
Removed